app/src/main/java/com/okta/oidc/example/OkHttp.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License,
+ * Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package com.okta.oidc.example;
+
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.WorkerThread;
+
+import com.okta.oidc.net.ConnectionParameters;
+import com.okta.oidc.net.OktaHttpClient;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.FormBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+
+/**
+ * A OktaHttpClient implementation using OkHttpClient.
+ */
+public class OkHttp implements OktaHttpClient {
+    private static final String TAG = "OkHttp";
+    /**
+     * The constant sOkHttpClient.
+     */
+    protected static OkHttpClient sOkHttpClient;
```

Comment:
Maybe it is better to provide this client as a construcor injection?

Reply:
This is a simple sample for my testing so for this sample I'm setting the connection parameters for the okhttpclient once for the instance. If I use a constructor the OkHttpClient is already created so the timeout cant be set because its done in a builder.